src/Config.php

```diff
@@ -124,14 +124,19 @@ public function setDefault($key, $value)
      * TODO: remove Data object validation in 2.0
      *
      * @return Data
+     * @throws \Exception
      */
     protected function getDefaults()
     {
         // Ensure $this->defaults is a Data object (not an array)
         if (is_array($this->defaults)) {
             $this->setDefaults($this->defaults);
+            return $this->getDefaults();
+        } elseif ($this->defaults instanceof Data) {
+            return $this->defaults;
+        } else {
```

Comment:
Minor style point: avoid `else` / extra indentation when unnecessary.

Reply:
I completely agree with both of these points and had a version working that didn't have either the `elseif` or the `else`. The problem is that Scrutinizer said there was a "major" error introduced because it couldn't detect that `$defaults` wasn't an array, and `array|Dflydev\DotAccessData\Data` didn't match the return type of `Dflydev\DotAccessData\Data`.

I would prefer the simpler version of the code, but also don't want to introduce problems that don't go away with the code checking that's in place.

Your recommendation for the final solution?

Reply:
I don't know if there is a solution to convince Scrutinizer that the return values are correct here, but it is permissible to ignore Scrutinizer inspection results if they are wrong.

Reply:
My previous code just checked if `$this->defaults` was an array, and if so set it using `$this->setDefaults($this->defaults)` then returned `$this->defaults` (now set as a `Data` object).

This is what was causing problems in Scrutinizer.

As a solution to not using `elseif` and `else` but still solving the Scrutinizer issue, I could use an `if` in place of the `elseif` and simply throw and `\Exception()` without the else, because both if statements `return` if they are true.

Correct?

Reply:
Yes, that's what I was advocating -- `if` in place of `elseif`. No change to logic of the function, only to the representation, to make it easier to read.

Reply:
Done